apps/src/sites/studio/pages/googleBlocklyWrapper.js

```diff
@@ -0,0 +1,191 @@
+  blocklyWrapper.wrapReadOnlyProperty('js');
+  blocklyWrapper.wrapReadOnlyProperty('modalBlockSpace');
+  blocklyWrapper.wrapReadOnlyProperty('Msg');
+  blocklyWrapper.wrapReadOnlyProperty('Names');
+  blocklyWrapper.wrapReadOnlyProperty('netsim_locale');
+  blocklyWrapper.wrapReadOnlyProperty('Procedures');
+  blocklyWrapper.wrapReadOnlyProperty('removeChangeListener');
+  blocklyWrapper.wrapReadOnlyProperty('RTL');
+  blocklyWrapper.wrapReadOnlyProperty('tutorialExplorer_locale');
+  blocklyWrapper.wrapReadOnlyProperty('useContractEditor');
+  blocklyWrapper.wrapReadOnlyProperty('useModalFunctionEditor');
+  blocklyWrapper.wrapReadOnlyProperty('Variables');
+  blocklyWrapper.wrapReadOnlyProperty('weblab_locale');
+  blocklyWrapper.wrapReadOnlyProperty('Workspace');
+  blocklyWrapper.wrapReadOnlyProperty('Xml');
+
+  blocklyWrapper.wrapSettableProperty('assetUrl');
+  blocklyWrapper.wrapSettableProperty('behaviorEditor');
+  blocklyWrapper.wrapSettableProperty('BROKEN_CONTROL_POINTS');
+  blocklyWrapper.wrapSettableProperty('BUMP_UNCONNECTED');
+  blocklyWrapper.wrapSettableProperty('HSV_SATURATION');
+  blocklyWrapper.wrapSettableProperty('JavaScript');
+  blocklyWrapper.wrapSettableProperty('readOnly');
+  blocklyWrapper.wrapSettableProperty('showUnusedBlocks');
+  blocklyWrapper.wrapSettableProperty('SNAP_RADIUS');
+  blocklyWrapper.wrapSettableProperty('typeHints');
+  blocklyWrapper.wrapSettableProperty('valueTypeTabShapeMap');
+
+  blocklyWrapper.getGenerator = function() {
+    return this.JavaScript;
+  };
+  blocklyWrapper.findEmptyContainerBlock = function() {}; // TODO
+  blocklyWrapper.BlockSpace = {};
+  blocklyWrapper.BlockSpace.EVENTS = {};
+  blocklyWrapper.BlockSpace.EVENTS.MAIN_BLOCK_SPACE_CREATED =
+    'mainBlockSpaceCreated';
+  blocklyWrapper.BlockSpace.EVENTS.EVENT_BLOCKS_IMPORTED = 'blocksImported';
+  blocklyWrapper.BlockSpace.EVENTS.BLOCK_SPACE_CHANGE = 'blockSpaceChange';
+  blocklyWrapper.BlockSpace.EVENTS.BLOCK_SPACE_SCROLLED = 'blockSpaceScrolled';
+  blocklyWrapper.BlockSpace.EVENTS.RUN_BUTTON_CLICKED = 'runButtonClicked';
+  blocklyWrapper.BlockSpace.onMainBlockSpaceCreated = () => {}; // TODO
+  blocklyWrapper.BlockSpace.createReadOnlyBlockSpace = () => {}; // TODO
+
+  blocklyWrapper.Block.prototype.getTitleValue =
+    blocklyWrapper.Block.prototype.getFieldValue;
+  blocklyWrapper.Block.prototype.setHSV = function(h, s, v) {
+    return this.setColour(h);
+  };
+
+  blocklyWrapper.Input.prototype.appendTitle = function(a, b) {
+    return this.appendField(a, b);
+  };
+
+  Object.defineProperty(blocklyWrapper, 'mainBlockSpace', {
+    get: function() {
+      return this.blockly_.mainWorkspace;
+    }
+  });
+  Object.defineProperty(blocklyWrapper, 'mainBlockSpaceEditor', {
+    get: function() {
+      return this.blockly_.mainWorkspace;
+    }
+  });
+  Object.defineProperty(blocklyWrapper, 'SVG_NS', {
+    get: function() {
+      return this.blockly_.utils.dom.SVG_NS;
+    }
+  });
```

Comment:
I was wondering why we needed to do these this way, and turns out it's because function name changed  Maybe worth a second helper function that can cope with this?  Or a comment explaining why?

Reply:
Documenting an offline decision for posterity:
Adapting the helper function to enable wrapping properties where the alias doesn't match the underlying property would look something like this:
```
this.aliasProperty = function(aliasName, propertyPath) {
  Object.defineProperty(this, aliasName, {
    get: function() {
      return propertyPath
        .split('.')
        .reduce((object, property) => object[property], this.blockly_);
    }
  });
};
blocklyWrapper.aliasProperty('SVG_NS', 'utils.dom.SVG_NS');
blocklyWrapper.aliasProperty('ALIGN_CENTRE', 'ALIGN_CENTRE');
```

Which we decided would be harder to understand/maintain than just having these three getters defined individually. I added a comment in the code explaining that they're also just wrapping read-only properties.